### ISSUE TITLE
Pixel 8 almost unusable on latest stable release  / Multiple crashes and screen stuttering

### ISSUE THREAD

**Issue by blizzardengle** [GrapheneOS/os-issue-tracker#2619]
I have to constantly restart my Pixel 8 because it eventually becomes unusable. The keyboard constantly locks up or crashes and multiple apps do the same. The launcher constantly crashes and just sliding down the app drawer will trigger a crash sometimes. The date and time bar will disappear as well. Now the screen just flashes on and off as I try to navigate around; it appears its crashing and recovering in a loop. Here are some but not all of the errors I have been able to record; sometimes the phones crashes and I cant copy the error.

For context I do have Google apps installed and Play Store and Play Services installed with as minimal permissions as I can. I also have removed the `sensors` permission from just about everything I have installed.

**UPDATE**: In my testing I have found that sometimes (rarely) I can click through the open app to what is behind it. The app is still open and drawn on the screen but my input is passing through it. I confirmed this by running a few apps until the error occurred, which in turn launched new apps. I double checked and where I had been interacting with the open app was the same location on the home screen of the new app that randomly opened. Again, all I saw was the actual app that I had open but my input was passing through it.

```
type: crash
osVersion: google/shiba/shiba:14/UD1A.230803.041/2023102300:user/release-keys
package: com.android.settings:34
process: com.android.settings
processUptime: 0 + 0 ms

signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
Abort message: 'GrContext is abandoned/device lost at start of CanvasContext::draw'

backtrace:
      #00 pc 000000000006eca4  /apex/com.android.runtime/lib64/bionic/libc.so (abort+164) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
      #01 pc 000000000064be80  /apex/com.android.art/lib64/libart.so (art::Runtime::Abort(char const*)+1456) (BuildId: 6ea39c7629f248392ceeb7f63efc9f64)
      #02 pc 0000000000044730  /apex/com.android.art/lib64/libbase.so (android::base::SetAborter(std::__1::function<void (char const*)>&&)::$_0::__invoke(char const*)+80) (BuildId: fe892b2afe1b21a706992b20886336f8)
      #03 pc 0000000000011fd4  /system/lib64/liblog.so (__android_log_assert+308) (BuildId: 7b2c8efcf19100c35d7cd5b687e6277a)
      #04 pc 0000000000292c04  /system/lib64/libhwui.so (android::uirenderer::renderthread::CanvasContext::draw(bool)+2580) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #05 pc 000000000029516c  /system/lib64/libhwui.so (std::__1::__function::__func<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0, std::__1::allocator<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0>, void ()>::operator()() (.__uniq.264041412789356548918088680803242235290.c303f2d2360db58ed70a2d0ac7ed911b)+1324) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #06 pc 00000000002840c0  /system/lib64/libhwui.so (android::uirenderer::WorkQueue::process()+624) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #07 pc 00000000002a66b4  /system/lib64/libhwui.so (android::uirenderer::renderthread::RenderThread::threadLoop()+404) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #08 pc 000000000001530c  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+284) (BuildId: 6536c636a18676bd5113fb0378052e1c)
      #09 pc 00000000000db93c  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+204) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
      #10 pc 0000000000070570  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
```

```
type: crash
osVersion: google/shiba/shiba:14/UD1A.230803.041/2023102300:user/release-keys
package: com.android.launcher3:34
process: com.android.launcher3
processUptime: 0 + 0 ms

signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
Abort message: 'GrContext is abandoned/device lost at start of CanvasContext::draw'

backtrace:
      #00 pc 000000000006eca4  /apex/com.android.runtime/lib64/bionic/libc.so (abort+164) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
      #01 pc 000000000064be80  /apex/com.android.art/lib64/libart.so (art::Runtime::Abort(char const*)+1456) (BuildId: 6ea39c7629f248392ceeb7f63efc9f64)
      #02 pc 0000000000044730  /apex/com.android.art/lib64/libbase.so (android::base::SetAborter(std::__1::function<void (char const*)>&&)::$_0::__invoke(char const*)+80) (BuildId: fe892b2afe1b21a706992b20886336f8)
      #03 pc 0000000000011fd4  /system/lib64/liblog.so (__android_log_assert+308) (BuildId: 7b2c8efcf19100c35d7cd5b687e6277a)
      #04 pc 0000000000292c04  /system/lib64/libhwui.so (android::uirenderer::renderthread::CanvasContext::draw(bool)+2580) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #05 pc 000000000029516c  /system/lib64/libhwui.so (std::__1::__function::__func<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0, std::__1::allocator<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0>, void ()>::operator()() (.__uniq.264041412789356548918088680803242235290.c303f2d2360db58ed70a2d0ac7ed911b)+1324) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #06 pc 00000000002840c0  /system/lib64/libhwui.so (android::uirenderer::WorkQueue::process()+624) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #07 pc 00000000002a66b4  /system/lib64/libhwui.so (android::uirenderer::renderthread::RenderThread::threadLoop()+404) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #08 pc 000000000001530c  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+284) (BuildId: 6536c636a18676bd5113fb0378052e1c)
      #09 pc 00000000000db93c  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+204) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
      #10 pc 0000000000070570  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
```

```
type: crash
osVersion: google/shiba/shiba:14/UD1A.230803.041/2023102300:user/release-keys
package: com.android.dialer:2900000
process: com.android.dialer
processUptime: 0 + 0 ms

signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
Abort message: 'GrContext is abandoned/device lost at start of CanvasContext::draw'

backtrace:
      #00 pc 000000000006eca4  /apex/com.android.runtime/lib64/bionic/libc.so (abort+164) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
      #01 pc 000000000064be80  /apex/com.android.art/lib64/libart.so (art::Runtime::Abort(char const*)+1456) (BuildId: 6ea39c7629f248392ceeb7f63efc9f64)
      #02 pc 0000000000044730  /apex/com.android.art/lib64/libbase.so (android::base::SetAborter(std::__1::function<void (char const*)>&&)::$_0::__invoke(char const*)+80) (BuildId: fe892b2afe1b21a706992b20886336f8)
      #03 pc 0000000000011fd4  /system/lib64/liblog.so (__android_log_assert+308) (BuildId: 7b2c8efcf19100c35d7cd5b687e6277a)
      #04 pc 0000000000292c04  /system/lib64/libhwui.so (android::uirenderer::renderthread::CanvasContext::draw(bool)+2580) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #05 pc 000000000029516c  /system/lib64/libhwui.so (std::__1::__function::__func<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0, std::__1::allocator<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0>, void ()>::operator()() (.__uniq.264041412789356548918088680803242235290.c303f2d2360db58ed70a2d0ac7ed911b)+1324) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #06 pc 00000000002840c0  /system/lib64/libhwui.so (android::uirenderer::WorkQueue::process()+624) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #07 pc 00000000002a66b4  /system/lib64/libhwui.so (android::uirenderer::renderthread::RenderThread::threadLoop()+404) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #08 pc 000000000001530c  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+284) (BuildId: 6536c636a18676bd5113fb0378052e1c)
      #09 pc 00000000000db93c  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+204) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
      #10 pc 0000000000070570  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
```

---

This error looks unrelated to the above but it's the latest I have been able to grab:

```
type: crash
osVersion: google/shiba/shiba:14/UD1A.230803.041/2023102300:user/release-keys
package: com.citi.citimobile:163
process: com.citi.citimobileprocessUptime: 0 + 0 ms
installer: com.android.vending

signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0000000000000028

backtrace: 
     #00 pc 000000000084c298  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::GetRandomMmapAddr()+24) 
     #01 pc 0000000000674bec  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::MemoryAllocator::AllocateChunk(unsigned long, unsigned long, v8::internal::Executability, v8::internal::Space*)+64) 
     #02 pc 000000000067a7a8  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::PagedSpace::Expand()+92) 
     #03 pc 000000000067f364  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::PagedSpace::RawSlowRefillLinearAllocationArea(int, v8::internal::AllocationOrigin)+432)
     #04 pc 000000000067f14c  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::PagedSpace::SlowRefillLinearAllocationArea(int, v8::internal::AllocationOrigin)+108) 
     #05 pc 00000000005e7aec  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::PagedSpace::AllocateRaw(int, v8::internal::AllocationAlignment, v8::internal::AllocationOrigin)+148)
     #06 pc 00000000005d774c  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::Heap::AllocateRaw(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment)+432)
     #07 pc 00000000005fecb8  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::Heap::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment)+40)
     #08 pc 00000000005fed60  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::Heap::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment)+40)
     #09 pc 00000000005d12e0  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::FactoryBase<v8::internal::Factory>::NewFixedArrayWithFiller(v8::internal::Handle<v8::internal::Map>, int, v8::internal::Handle<v8::internal::Oddball>, v8::internal::AllocationType)+76)
     #10 pc 0000000000ec460c  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::Handle<v8::internal::FixedArray> v8::internal::interpreter::ConstantArrayBuilder::ToFixedArray<v8::internal::Isolate>(v8::internal::Isolate*)+108)
     #11 pc 0000000000ec02c4  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::Handle<v8::internal::BytecodeArray> v8::internal::interpreter::BytecodeArrayWriter::ToBytecodeArray<v8::internal::Isolate>(v8::internal::Isolate*, int, int, v8::internal::Handle<v8::internal::ByteArray>)+56)
     #12 pc 0000000000b54cb8  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::Handle<v8::internal::BytecodeArray> v8::internal::interpreter::BytecodeGenerator::FinalizeBytecode<v8::internal::Isolate>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Script>)+292) 
     #13 pc 00000000006a3774  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::CompilationJob::Status v8::internal::interpreter::InterpreterCompilationJob::DoFinalizeJobImpl<v8::internal::Isolate>(v8::internal::Handle<v8::internal::SharedFunctionInfo>, v8::internal::Isolate*)+772)
     #14 pc 00000000006a33f0  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::interpreter::InterpreterCompilationJob::FinalizeJobImpl(v8::internal::Handle<v8::internal::SharedFunctionInfo>, v8::internal::Isolate*)+304)
     #15 pc 00000000005641e8  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::(anonymous namespace)::FinalizeUnoptimizedCompilationJob(v8::internal::UnoptimizedCompilationJob*, v8::internal::Handle<v8::internal::SharedFunctionInfo>, v8::internal::Isolate*)+112)
     #16 pc 000000000055e27c  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (bool v8::internal::(anonymous namespace)::FinalizeUnoptimizedCode<v8::internal::Isolate>(v8::internal::ParseInfo*, v8::internal::Isolate*, v8::internal::Handle<v8::internal::SharedFunctionInfo>, v8::internal::UnoptimizedCompilationJob*, std::__ndk1::forward_list<std::__ndk1::unique_ptr<v8::internal::UnoptimizedCompilationJob, std::__ndk1::default_delete<v8::internal::UnoptimizedCompilationJob> >, std::__ndk1::allocator<std::__ndk1::unique_ptr<v8::internal::UnoptimizedCompilationJob, std::__ndk1::default_delete<v8::internal::UnoptimizedCompilationJob> > > >*)+208)
     #17 pc 000000000055dd64  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::Compiler::Compile(v8::internal::Handle<v8::internal::SharedFunctionInfo>, v8::internal::Compiler::ClearExceptionFlag, v8::internal::IsCompiledScope*)+1124)
     #18 pc 000000000055e42c  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::Compiler::Compile(v8::internal::Handle<v8::internal::JSFunction>, v8::internal::Compiler::ClearExceptionFlag, v8::internal::IsCompiledScope*)+292)
     #19 pc 0000000000ba65f8  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (v8::internal::Runtime_CompileLazy(int, unsigned long*, v8::internal::Isolate*)+164)
     #20 pc 0000000000a0a128  /data/app/~~FAvHTGTMzQzTCFdzrs3gqA==/com.citi.citimobile-Z-pkgkctlI7-tfI8P9MEkQ==/lib/arm64/libj2v8.so (Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_NoBuiltinExit+104)
```

**Comment by blizzardengle** [comment]
[This issue](https://issuetracker.google.com/issues/255397162?pli=1) may be related or the same bug. I notice if I use the phone to fast it triggers these issues quicker just like this comment:

> This crash is fairly easy to reproduce:
>
> Open a webview and target something that is easy to do fast back and forth navigation on (example: gmail from the web, entering and exiting emails). Navigate using on screen back buttons embedded into the site (not the device back button).
>
> If you navigate your webview back and forth repeatedly (and quickly) this crash should happen within 50 navigations or so. Sometimes less than 10 navigations.
> ([context](https://issuetracker.google.com/issues/255397162#comment27))

**Comment by emacsomancer** [comment]
This seems connected with https://github.com/GrapheneOS/os-issue-tracker/issues/2601

**Comment by blizzardengle** [comment]
> This seems connected with https://github.com/GrapheneOS/os-issue-tracker/issues/2601

I don't know if they are. They cause a similar problem (symptom) but the abort messages are very different and in my case the crash can be caused by any app not just the launcher.

I have not been able to pin point what the initial trigger is but I have noticed a new issue:

**UPDATE**: In my testing I have found that sometimes (rarely) I can click through the open app to what is behind it. The app is still open and drawn on the screen but my input is passing through it. I confirmed this by running a few apps until the error occurred, which in turn launched new apps. I double checked and where I had been interacting with the open app was the same location on the home screen of the new app that randomly opened. Again, all i saw was the actual app that i had open but my input pass through it.

**Comment by emacsomancer** [comment]
I think there are a cluster of possibly inter-related things. It's not just the launcher which seems to trigger crashes. Other things do as well, including live wallpapers.

Another crashlog I just got:

```
type: crash
osVersion: google/raven/raven:14/UP1A.231005.007/2023102300:user/release-keys
package: com.android.launcher3:34
process: com.android.launcher3
processUptime: 2062 + 307 ms

android.os.DeadSystemRuntimeException: android.os.DeadSystemException
	at android.content.pm.LauncherApps.getApplicationInfo(LauncherApps.java:1081)
	at android.content.pm.LauncherApps.loadDrawableResourceFromPackage(LauncherApps.java:1564)
	at android.content.pm.LauncherApps.getShortcutIconDrawable(LauncherApps.java:1470)
	at com.android.launcher3.icons.ShortcutCachingLogic.getIcon(ShortcutCachingLogic.java:109)
	at com.android.launcher3.icons.ShortcutCachingLogic.loadIcon(ShortcutCachingLogic.java:77)
	at com.android.launcher3.icons.ShortcutCachingLogic.loadIcon(ShortcutCachingLogic.java:43)
	at com.android.launcher3.icons.cache.BaseIconCache.addIconToDBAndMemCache(BaseIconCache.java:324)
	at com.android.launcher3.icons.cache.IconCacheUpdateHandler$SerializedIconUpdateTask.run(IconCacheUpdateHandler.java:277)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.os.HandlerThread.run(HandlerThread.java:67)
Caused by: android.os.DeadSystemException
	... 13 more
```

**Comment by blizzardengle** [comment]
This error just happened trying to use the GitHub app:

```
type: crash
osVersion: google/shiba/shiba:14/UD1A.230803.041/2023102300:user/release-keys
package: com.github.android:754
process: com.github.android
processUptime: 0 + 0 ms
installer: com.android.vending

signal 5 (SIGTRAP), code 1 (TRAP_BRKPT), fault addr 0x0000d1f593e28f2c
Abort message: '[FATAL:aw_draw_fn_impl.cc(113)] Failed GrVkSecondaryCBDrawContext::Make fSecondaryCommandBuffer:0xb400d2d801eeed00 fColorAttachmentIndex:0 fCompatibleRenderPass:0xb400d278167df580 fFormat:37 width:1080 height:2400 is_srgb:1'

backtrace:
      #00 pc 00000000045f3f2c  /data/app/~~sDO8fw1RoqxfLBtpMAB3Ig==/app.vanadium.trichromelibrary_599311132-csEv6kywYBchddcEZ4wDJQ==/base.apk!libmonochrome_64.so (offset 0x8c0000) (BuildId: 1177265ed36cdcdf99913fee6068aa12d477271b)
      #01 pc 000000000220bd30  /data/app/~~sDO8fw1RoqxfLBtpMAB3Ig==/app.vanadium.trichromelibrary_599311132-csEv6kywYBchddcEZ4wDJQ==/base.apk!libmonochrome_64.so (offset 0x8c0000) (BuildId: 1177265ed36cdcdf99913fee6068aa12d477271b)
      #02 pc 0000000000010738  /system/lib64/libwebviewchromium_plat_support.so (android::(anonymous namespace)::drawVk(int, void*, android::uirenderer::VkFunctorDrawParams const&, android::uirenderer::WebViewOverlayData const&)+360) (BuildId: 92a13b63494a04d8a28ea92013f438b7)
      #03 pc 00000000002b59a4  /system/lib64/libhwui.so (android::uirenderer::WebViewFunctor::drawVk(android::uirenderer::VkFunctorDrawParams const&)+180) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #04 pc 000000000028eb3c  /system/lib64/libhwui.so (android::uirenderer::skiapipeline::VkFunctorDrawHandler::draw(GrBackendDrawableInfo const&)+332) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #05 pc 00000000006ad7a8  /system/lib64/libhwui.so (GrVkOpsRenderPass::onExecuteDrawable(std::__1::unique_ptr<SkDrawable::GpuDrawHandler, std::__1::default_delete<SkDrawable::GpuDrawHandler> >)+248) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #06 pc 00000000005a2edc  /system/lib64/libhwui.so (GrOpsRenderPass::executeDrawable(std::__1::unique_ptr<SkDrawable::GpuDrawHandler, std::__1::default_delete<SkDrawable::GpuDrawHandler> >)+60) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #07 pc 00000000006593a4  /system/lib64/libhwui.so (skgpu::v1::DrawableOp::onExecute(GrOpFlushState*, SkRect const&)+52) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #08 pc 000000000066da00  /system/lib64/libhwui.so (skgpu::v1::OpsTask::onExecute(GrOpFlushState*)+1040) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #09 pc 000000000058f94c  /system/lib64/libhwui.so (GrDrawingManager::flush(SkSpan<GrSurfaceProxy*>, SkSurface::BackendSurfaceAccess, GrFlushInfo const&, skgpu::MutableTextureState const*)+2844) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #10 pc 000000000058fee0  /system/lib64/libhwui.so (GrDrawingManager::flushSurfaces(SkSpan<GrSurfaceProxy*>, SkSurface::BackendSurfaceAccess, GrFlushInfo const&, skgpu::MutableTextureState const*)+160) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #11 pc 00000000005892ec  /system/lib64/libhwui.so (GrDirectContextPriv::flushSurfaces(SkSpan<GrSurfaceProxy*>, SkSurface::BackendSurfaceAccess, GrFlushInfo const&, skgpu::MutableTextureState const*)+268) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #12 pc 00000000006c85d4  /system/lib64/libhwui.so (SkSurface_Gpu::onFlush(SkSurface::BackendSurfaceAccess, GrFlushInfo const&, skgpu::MutableTextureState const*)+148) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #13 pc 00000000002996ac  /system/lib64/libhwui.so (android::uirenderer::renderthread::VulkanManager::finishFrame(SkSurface*)+284) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #14 pc 000000000028e3b0  /system/lib64/libhwui.so (android::uirenderer::skiapipeline::SkiaVulkanPipeline::draw(android::uirenderer::renderthread::Frame const&, SkRect const&, SkRect const&, android::uirenderer::LightGeometry const&, android::uirenderer::LayerUpdateQueue*, android::uirenderer::Rect const&, bool, android::uirenderer::LightInfo const&, std::__1::vector<android::sp<android::uirenderer::RenderNode>, std::__1::allocator<android::sp<android::uirenderer::RenderNode> > > const&, android::uirenderer::FrameInfoVisualizer*, android::uirenderer::renderthread::HardwareBufferRenderParams const&)+400) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #15 pc 00000000002924f0  /system/lib64/libhwui.so (android::uirenderer::renderthread::CanvasContext::draw(bool)+768) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #16 pc 000000000029516c  /system/lib64/libhwui.so (std::__1::__function::__func<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0, std::__1::allocator<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0>, void ()>::operator()() (.__uniq.264041412789356548918088680803242235290.c303f2d2360db58ed70a2d0ac7ed911b)+1324) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #17 pc 00000000002840c0  /system/lib64/libhwui.so (android::uirenderer::WorkQueue::process()+624) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #18 pc 00000000002a66b4  /system/lib64/libhwui.so (android::uirenderer::renderthread::RenderThread::threadLoop()+404) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #19 pc 000000000001530c  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+284) (BuildId: 6536c636a18676bd5113fb0378052e1c)
      #20 pc 00000000000db93c  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+204) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
      #21 pc 0000000000070570  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
```

**Comment by thestinger** [comment]
None of this can be replicated by the development team. There's something quite rare about your setup causing it.

**Comment by blizzardengle** [comment]
@thestinger The issue was so bad I just rest my phone, it was truly unusable. I installed Brave and switched web view (preview?) to it instead of Vanadium. I doubt that was the issue but maybe. I will install Google apps slowly over a few days and see if this happens again.

**Comment by blizzardengle** [comment]
@thestinger never mind. Brand new install with barely any changes to the phone. Setup pin lock and fingerprint lock. Enabled dark theme. Tried to install Google Play store and Google Service and crash:

```
type: crash
osVersion: google/shiba/shiba:14/UD1A.230803.041/2023102300:user/release-keys
package: app.grapheneos.apps:21
process: app.grapheneos.apps
processUptime: 0 + 0 ms

signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
Abort message: 'GrContext is abandoned/device lost at start of CanvasContext::draw'

backtrace:
      #00 pc 000000000006eca4  /apex/com.android.runtime/lib64/bionic/libc.so (abort+164) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
      #01 pc 000000000064be80  /apex/com.android.art/lib64/libart.so (art::Runtime::Abort(char const*)+1456) (BuildId: 6ea39c7629f248392ceeb7f63efc9f64)
      #02 pc 0000000000044730  /apex/com.android.art/lib64/libbase.so (android::base::SetAborter(std::__1::function<void (char const*)>&&)::$_0::__invoke(char const*)+80) (BuildId: fe892b2afe1b21a706992b20886336f8)
      #03 pc 0000000000011fd4  /system/lib64/liblog.so (__android_log_assert+308) (BuildId: 7b2c8efcf19100c35d7cd5b687e6277a)
      #04 pc 0000000000292c04  /system/lib64/libhwui.so (android::uirenderer::renderthread::CanvasContext::draw(bool)+2580) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #05 pc 000000000029516c  /system/lib64/libhwui.so (std::__1::__function::__func<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0, std::__1::allocator<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0>, void ()>::operator()() (.__uniq.264041412789356548918088680803242235290.c303f2d2360db58ed70a2d0ac7ed911b)+1324) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #06 pc 00000000002840c0  /system/lib64/libhwui.so (android::uirenderer::WorkQueue::process()+624) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #07 pc 00000000002a66b4  /system/lib64/libhwui.so (android::uirenderer::renderthread::RenderThread::threadLoop()+404) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #08 pc 000000000001530c  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+284) (BuildId: 6536c636a18676bd5113fb0378052e1c)
      #09 pc 00000000000db93c  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+204) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
      #10 pc 0000000000070570  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
```

**Comment by thestinger** [comment]
Your phone seems to have a hardware issue. I can't think of another explanation.

**Comment by blizzardengle** [comment]
> Your phone seems to have a hardware issue. I can't think of another explanation.

**TL;DR** I disagree since the stock os works, but I understand that its a edge case for sure.

I used the phone with the stock OS for a while before factory resetting it and putting Graphene on it. I just reverted back to the stock OS and after a 30 minute stress test I can't reproduce these issues on stock.

I don't know why this only happens on Graphene. My educated guess is maybe I have an earlier production run model with slight different hardware? Either way I am returning the phone. I have a Pixel 6a that has been running Graphene flawlessly for months with the exact same setup apps/permissions wise; for the curious, no I did not load a backup, I manually setup all the apps/permissions on the Pixel 8.

@thestinger I'm leaving this issue open for now just in case someone else runs into this problem, but since I'm returning the phone I understand if you need to mark it closed.

**Comment by thestinger** [comment]
Doesn't make any sense since you're the only one with the issues and you say it's on a fresh install of GrapheneOS.

**Comment by thestinger** [comment]
@blizzardengle If you install GrapheneOS again and **don't install a bunch of apps, make configuration changes, etc.** do you have the problem?

**Comment by blizzardengle** [comment]
@thestinger yes, but here is the full story:

Since everything appeared good on stock, I factory reset, and then setup everything (apps/permissions) right out of the gate on Graphene and noticed these issues within the first 4-6hrs of use I would say. Then the phone just became unusable. I would have to restart it after 10 minutes of use.

To double check everything, I factory reset again, and used Graphene stock for a few minutes; opened Settings, App, File, and Calculator. Couldn't trigger the issue. As soon I started to install Google Play Store and Google Play Services it started happening again; which was weird to me because I hadn't even tried to install a Play store app yet and previously it didn't happen that fast. That was this crash:

```
type: crash
osVersion: google/shiba/shiba:14/UD1A.230803.041/2023102300:user/release-keys
package: app.grapheneos.apps:21
process: app.grapheneos.apps
processUptime: 0 + 0 ms

signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
Abort message: 'GrContext is abandoned/device lost at start of CanvasContext::draw'

backtrace:
      #00 pc 000000000006eca4  /apex/com.android.runtime/lib64/bionic/libc.so (abort+164) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
      #01 pc 000000000064be80  /apex/com.android.art/lib64/libart.so (art::Runtime::Abort(char const*)+1456) (BuildId: 6ea39c7629f248392ceeb7f63efc9f64)
      #02 pc 0000000000044730  /apex/com.android.art/lib64/libbase.so (android::base::SetAborter(std::__1::function<void (char const*)>&&)::$_0::__invoke(char const*)+80) (BuildId: fe892b2afe1b21a706992b20886336f8)
      #03 pc 0000000000011fd4  /system/lib64/liblog.so (__android_log_assert+308) (BuildId: 7b2c8efcf19100c35d7cd5b687e6277a)
      #04 pc 0000000000292c04  /system/lib64/libhwui.so (android::uirenderer::renderthread::CanvasContext::draw(bool)+2580) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #05 pc 000000000029516c  /system/lib64/libhwui.so (std::__1::__function::__func<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0, std::__1::allocator<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0>, void ()>::operator()() (.__uniq.264041412789356548918088680803242235290.c303f2d2360db58ed70a2d0ac7ed911b)+1324) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #06 pc 00000000002840c0  /system/lib64/libhwui.so (android::uirenderer::WorkQueue::process()+624) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #07 pc 00000000002a66b4  /system/lib64/libhwui.so (android::uirenderer::renderthread::RenderThread::threadLoop()+404) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #08 pc 000000000001530c  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+284) (BuildId: 6536c636a18676bd5113fb0378052e1c)
      #09 pc 00000000000db93c  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+204) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
      #10 pc 0000000000070570  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
```

Factory reset again, reverted to stock android, and stressed tested for 30 minutes. I installed tons of apps, opened every app I could, typed into multiple keyboards, force killed apps, ect. I could not reproduce this on the stock os.

**Comment by thestinger** [comment]
You're doing something rare other people aren't doing that's causing it. If you won't narrow that down, we can't help.

**Comment by thestinger** [comment]
If you narrow down what's causing it, perhaps we can resolve it.

**Comment by blizzardengle** [comment]
I don't know how to narrow that down. Here is a full breakdown roughly in order:

**Attempt 1: Initial Install of Graphene**

- Factory rest from stock; I had partially set it up so I could make sure the phone worked, no sim card.
- Installed GrapheneOS via website.
- Connected to WiFi only; no sim yet.
- Updated stock Graphene apps.
- Installed Google services:
    - GmsCompactConfig
    - Google Services Framework
    - Google Play Store
    - REVOKED sensor permissions form all of them; I think I later allowed sensors for GmsCompactConfig
- Installed various Google apps, setting them up one by one, revoking the sensor permission from everything:
    - Mint Mobile; needed to fully setup my sim. I restarted my phone and added the sim at this point.
    - Aegis for my OTP codes. Setup by scaning QR code from my Pixel 6a running Graphene.
    - Duo; needed for work, copied settings via QR code from my Pixel 6a running Graphene.
    - Facebook Messenger
    - Gmail
    - Google Contacts
    - Google Calendar
        -  Setup the calendar widget for it on one whole screen.
    - Voice
    - Photos
        - Storage scoped to a Google folder I made and the Camera folder.
    - Google Camera
        - Storage scoped to a Google folder I made and the Camera folder.
    - Voice Recorder
        - Including the Speech Recognition and Synthesis from Google which I revoked Network access from after downloading the English language pack.
        - I did leave network on for Voice so I can backup recordings to Google.
    - Outlook
        -  Setup the calendar widget for it on one whole screen.
    - OneDrive
    - GitHub App
    - Messages by Google; never used before but figured I would try, allowed setting up RCS texting.
- At this point I started using the phone:
    - Made a long call to my mom to test the default Graphene Phone.
    - Texting people back via Voice and Messages.
    - Used Facebook Messenger to message friends.
    - Changed/setup basic settings:
        - Enabled dark theme.
        - Left haptic feedback on but turned off Touch feedback vibrations.
        - Disabled Tap and Click sounds.
- Sometime near the end of this list I started to notice screen freezes. Then it would flash or go black like something had crashed and restarted. Eventually the crash messages started showing up. I restarted my phone and the issue went away for maybe 30 minutes; I was barely using the phone at that time. When I use the phone fast, quickly opening, switching, and closing things it happened a lot quicker. I have a habit of pulling the notification drawer down as soon as a notification appears, mid apps loading, or mid typing, and that is when some of the crashes I was able to report appeared.

---

**Attempt 2: Clean Install of Graphene**

- Factory rest from Graphene; sim card was left in this time.
- Installed GrapheneOS via website.
- Connected to WiFi during setup.
- Updated stock Graphene apps.
- Used the phone for a bit without any major changes.
    - Played with the built in App, File, and Calculator.
- Changed/setup basic settings:
    - Enabled dark theme.
    - Left haptic feedback on but turned off Touch feedback vibrations.
    - Disabled Tap and Click sounds.
- Attempted to install Google services:
    - GmsCompactConfig
    - Google Services Framework
    - Google Play Store
- Just as I started the install I remembered I left the apps open I was playing with and switched away to close them. In the process of closing them I got this error:

```
type: crash
osVersion: google/shiba/shiba:14/UD1A.230803.041/2023102300:user/release-keys
package: app.grapheneos.apps:21
process: app.grapheneos.apps
processUptime: 0 + 0 ms

signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
Abort message: 'GrContext is abandoned/device lost at start of CanvasContext::draw'

backtrace:
      #00 pc 000000000006eca4  /apex/com.android.runtime/lib64/bionic/libc.so (abort+164) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
      #01 pc 000000000064be80  /apex/com.android.art/lib64/libart.so (art::Runtime::Abort(char const*)+1456) (BuildId: 6ea39c7629f248392ceeb7f63efc9f64)
      #02 pc 0000000000044730  /apex/com.android.art/lib64/libbase.so (android::base::SetAborter(std::__1::function<void (char const*)>&&)::$_0::__invoke(char const*)+80) (BuildId: fe892b2afe1b21a706992b20886336f8)
      #03 pc 0000000000011fd4  /system/lib64/liblog.so (__android_log_assert+308) (BuildId: 7b2c8efcf19100c35d7cd5b687e6277a)
      #04 pc 0000000000292c04  /system/lib64/libhwui.so (android::uirenderer::renderthread::CanvasContext::draw(bool)+2580) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #05 pc 000000000029516c  /system/lib64/libhwui.so (std::__1::__function::__func<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0, std::__1::allocator<android::uirenderer::renderthread::DrawFrameTask::postAndWait()::$_0>, void ()>::operator()() (.__uniq.264041412789356548918088680803242235290.c303f2d2360db58ed70a2d0ac7ed911b)+1324) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #06 pc 00000000002840c0  /system/lib64/libhwui.so (android::uirenderer::WorkQueue::process()+624) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #07 pc 00000000002a66b4  /system/lib64/libhwui.so (android::uirenderer::renderthread::RenderThread::threadLoop()+404) (BuildId: 56a6ea38174b012e1e73074e7c0a4b2e)
      #08 pc 000000000001530c  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+284) (BuildId: 6536c636a18676bd5113fb0378052e1c)
      #09 pc 00000000000db93c  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+204) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
      #10 pc 0000000000070570  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64) (BuildId: 70e39a6810cd2a8f7b6f23f9b5ea610d)
```

This was a clean install of Graphene. I restarted all the installs for Google Play Store and Services after dismissing the error and NEVER removed the sensor permission from anything like I had done in the first attempt. I played with the phone for a bit and issues started manifesting again even without installing a single Google app. I think I installed Gmail and Voice after this to keep testing and all the errors/crashes were present again.

---

**Attempt 3: Clean Install of Stock**

- Factory reset from Graphene.
- Removed the boot key via Graphene's website.
- Reinstalled stock via Google's website.
- Stress tested for 30 minutes trying to force this error to happen. No matter what I did I never could get the error to happen. Stock works perfectly which leads me to think this is a software (albeit edge case) issue and not a hardware issue.

---

This is honesty the best I can do to help.

I have a bug report recorded by the phone when I was using Graphene if you have a private email I can send it to?

**Comment by blizzardengle** [comment]
**UPDATES**

After getting some food and refocusing I had these thoughts:

1. I used Brave browser to do the install of Graphene last time. Maybe try a different browser.
2. I can't be sure but I installed Brave browser on Graphene and made it my default WebView app before or around the time these issues started. Maybe don't do that.
3. I went crazy with revoking sensor permissions for everything. Maybe be more careful revoking that permission; I was only focusing on Google apps and not Graphene but still.
4. I did try to pair a Pixel Watch 2 before or around the time these issues started. Maybe that corrupted something? But why did a factory reset of Graphene still persist the issues?

With this in mind I decided to track every little thing I do to see if I could replicate this issue. I flashed the stock os again, made sure the phone was working, and then restarted from scratch the install of Graphene, but this time:

1. Used Edge to install Graphene just in case Brave is the issue. I am using an enterprise controlled laptop and installs of apps are allowed but Brave is not the standard or supported work browser.
2. Do not install Brave on the phone or attempt to change the WebView; just eliminate this possibility for now by not trying it.
3. Very carefully revoked sensor permissions from apps this time.
4. Do not even attempt to pair a watch or even install the Watch app that needs a lot of permissions.

**TL;DR** I moved forward with this plan and tracked everything. This time no matter what I did I could not trigger the crashes anymore, but I occasionally had hiccups. It should be noted that I have not installed all the other apps that I had last time. I am probably only half way through with the apps. I will keep using for a few days before installing anymore apps. What I have is the core I need so we'll see how it goes.

**Debug Session**

- Removed phone case.
- Starting with Stock OS.
- Flashed Graphene using Edge Brower insetad of Brave; Windows 11 Enterprise, partially locked down by my employer.
- Setup:
	- Language: English US
	- Date & Time: GMT-0:600
	- WiFi: Connected to my home network (802.11ax)
	- No Sim, skipped.
	- Location Services: Allowed
	- Pin and Fingerprint setup; setup both hands.
	- Restore Apps and Date: Skipped.
- Played with Calculator, Files, Gallery, and Camera app. Took a few pictures and deleted them via Files. Left everything open.
- Opened Apps app and installed updates: Vanadium and Vanadium System WebView.
	- Played with other apps as soon as the download started.
	- Closed all apps well install was still happening, including closing Apps itself.
- Enabled developer mode.
- Diabled OEM unblocking.
- [NEW] Disabled window-level blurs; I hear it helps save battery and I wonder if it may be part of the issue.
- Disabled developer mode and restrated phone.
- [Issue?] Had difficulting getting the lock screen to open the pin pad; it would seem to pop up and then flash and show the lock again x 2.
- Played with apps again:
	- Calculator: All good.
	- Messaging: [Error] "This app was built for an older version of Android... Check for an update2, or contact the app's developer". Checked for update, none was found. Closed app.
	- Vanadium: Allowed notifications. Tested some websites. Typed into textareas, no keyboard issues.
	- Camera: Took some pictures and closed app.
	- Closed all open apps.
- Settings: Enabled dark theme.
- Played with apps again.
- Disabled location services.
- Settings: Disabled dark theme; I have a feeling this may be part of the issue since I always enabled in the past.
- Closed all apps.
- Started the install of Google Play Services, immediatly closed Apps. Installs worked and I was prompted to allow network access. I approved 1 of 3 and then closed Apps again. No error or crash but I had to install the apps again. As the apps are installing I closed Apps again; no apps are running, I am just staring at the home screen logging this.
- Reopened Apps and all the Google Services are installed.
- Open Play Store:
	- Disabled the "Sandbox Google Play is Running" notification.
	- Allowed Play Services to always run in the background.
	- Allowed Play Store to show notifications.
- Cleared all notifications and closed all apps.
- Opened Play Store and signed in.
	- Allowed Play Services Contact Permission.
- Played with apps again: Camera, Calculator, Files, and Vanadium; left them all running.
- Shutdown phone, installed my sim card.
- Pin pad opened fine this time.
- 3 Apps optimized. Approved app restart.
- Settings: Enabled pin scramble.
- Settings: Lowers auto reboot to 18 hours.
- Closed all open apps that I had left open from last session.
- Restarted the phone to check that pin scramble is working.
	- It works and pin pad loaded no issues.
- [Why?] App restart is required notification again.
- Text message to setup Mint Sim.
- Play Store installed Mint Mobile app:
	- Had to allow Play Store to install apps.
	- Had to allow network access for Mint.
	- Opened Mint and setup my account.
- Play Store installed Messages and ignore network permission box to force error -> Can't install "send feedback" or "got it". Clicked "got it" and reinstalled.
	- Setup Storage Scope and allowed folders "Pixel 8/Scopes/Google" and "Pixel 8/DCIM/Camera" for Music and Audio + Photos and Videos.
	- Revoked sensors permission.
	- Allowed setup of RCS chats.
- Play Store installed Google Voice.
	- Microphone and Camera allowed when app is in use.
	- Revoked sensors permission.
- Texting myself back and forth between Voice and Messages.
- Settings: Enabled dark theme.
- Texted myself again.
- Play Store installed Pixel Camera and Google Photos.
	- Setup Storage Scope for both apps and allowed folders "Pixel 8/Scopes/Google" and "Pixel 8/DCIM/Camera".
	- Revoked sensors from Photos, allowed it for Camera.
	- Denied both Location.
- Played with all the apps and restarted the phone to check pin pad again.
	- Again, pin pad loaded fine no issue this time. Still confused why it had hicups the first restart.
= Added phone case.
- Settings:
	- Left haptics on but turned off Touch Feedback.
	- Diabled (turned off) Tap & Click sounds.
- Play Store installed Gmail:
	- Setup Storage Scope and allowed folders "Pixel 8/Scopes/Google" and "Pixel 8/DCIM/Camera".
	- Revoked sensors.
- Revoked Play Store sensors permission.
- Play Store installed Google Calendar:
	- Revoked sensors.
	- Allowed contacts and calendar.
- [Issue?] Added Google Calendar widget to second screen:
	- Did not place first time. Screen flashed and nothing placed, still only one screen.
	- Attempt 2 worked. Calander widget is fullscreen on screen 2.
- Play Store installed Outlook.
	- Revoked sensors.
	- Setup Storage Scope and allowed folders "Pixel 8/Scopes/Google" and "Pixel 8/DCIM/Camera".
	- Allowed notifications but no other permissions.
- Added Outlook Calander fullscreen on screen 3. No issues.
- Closed all open apps: I had left like 10 open at this point.
- Settings: Bumped up font size and display size by 1 each.
- [New] Play Story installed Memeber Tools and revoked sensors permission.
- Play Store installed Duo and Aegis.
	- Aegis: Installed icon pack from zip download and configured.
- Reorganized tiles.
- Setup alarms for the morning.

**Comment by blizzardengle** [comment]
I can not pin point what the root cause is. Using stock Graphene with no other apps rarely triggers this issue, it is only after installing  Google Services, Framework, and Play Store that the issue becomes more noticeable; and this is without an other apps added yet. Adding only 7 apps and testing here is my observations:

- With no Google apps installed besides  Google Services, Framework, and Play Store the crashes can and do occur but rarely.
- The more apps installed to the phone via Play Store the faster issues arise to the point the phone is unusable.
- Memory usage seems off:
        - Android OS uses > 2gb all the time which may be normal on a Pixel 8 but my 6a uses < 1.5gb.
        - Google Play Services is consistently using 400mb or more even when not in use and with all apps closed. My Pixel 6a uses < 200mb with 5 times as many apps installed.
        - System UI and Launcher are also double what my  Pixel 6a uses.
- Based on my limited understanding of the bug reports I have run I think a threaded process is being orphaned or not shut down correctly and the longer the phone is used and the more apps on the phone the faster I start running into stability issues:
        - Screen tear/screen artifacts/crashing and restarting apps/above average memory usage.

**Debug Session**

- Factory Rest Graphene.
- Setup:
	- Language: English US.
	- Date & Time: GMT-0:600
	- WiFi: Connected to my home network (802.11ax)
	- NOTE: Sim is installed.
	- Location Services: Allowed.
	- Pin and Fingerprint setup; setup both hands.
	- Restore Apps and Date: Skipped.
- Updated Apps.
- Stressed tested phone for 30 minutes with only Stock Apps; sent a lot of messages of verying lengths, took random pictures.
- Installed Google Services, Framework, and Play Store.
- Stressed tested phone for 30 minutes with only Stock Apps; sent a lot of messages of verying lengths, took random pictures.
	- No crash reports but I feel like I experienced a "soft crash" where Play Store was relaunched instantly from a crash.
- Login to Play Store.
- Restart phone to check pin pad; it glitched before so I wanted to check early in this test. No issues.
- Play Store installed:
	- Gmail
	- Calendar
	- Voice
	- Contacts
	- Messages
	- Photos
	- Camera by Google
- Stressed tested phone for ~45 minutes. I sent a lot of messages of verying lengths from Voice and Messages, took random pictures, synced them up with Photos, edited contacts, etc.
	- Again, no major crashes or screen tear/stutter/lock up like before but various apps would appear to "soft crash" by flashing. Did not happen a lot, maybe I am just paranoid at this point.
- Set alarms, went to bed.
- Alarms went off no issues.
- Started using the phone and sure enough crashes happened again as I used the apps; which yes at this point are all Google apps, but the System Lanucher keeps crashing too.
- Conclusion based on past observations:
	- Time is part of the issue. After 3-4 hours of uptime the phone would become unusable in the past; I had a ton of apps installed.
	- Time is longer now with only a few apps. I made it to ~8 hours uptime before the phone started having an annoying level of crashes and needed to be restarted.

**Closing Thoughts**

Maybe I got a lemon. I will keep the phone for 1 more day in case anyone has specific ideas of what to test, but tomorrow I'm flashing stock back on and returning it. **Aside:** I have abused and stressed tested my Pixel 6a with Graphene for ~3 months with no issues. If I have some *rare* workflow surely I would have bricked that phone long ago.

**Comment by mattinahat13** [comment]
Hi all, I am running into this same issue where the phone will freeze and require a hard reset. It mostly happens while I am on a walk where I will be playing music/podcasts over bluetooth and have my flashlight on, but it has happened randomly as well. Any app that is open (even settings, launcher, and systemui) will crash and include this in the error message:

Abort message: 'GrContext is abandoned/device lost at start of CanvasContext::draw'.

Looking through the comments, I will note that I do not have GPlay services/store installed, but I did have them at one point to download an eSIM.

This issue doesn't occur on my Pixel 7Pro.

**Comment by blizzardengle** [comment]
> Hi all, I am running into this same issue where the phone will freeze and require a hard reset...
>
> Abort message: 'GrContext is abandoned/device lost at start of CanvasContext::draw'.
>
> Edit: The crash will often turn off my flashlight, but not the music playing.

@mattinahat13 I'm sorry to hear your having the same issue but I'm relieved to know its not just my phone. I can confirm my crashes have been similar to yours; music will still play for example.

I really think there might be some regression issue related/similar to [this graphics bug](https://issuetracker.google.com/issues/255397162?pli=1) in Android. Maybe some phones have different versions (iterations) of the hardware like the GPU.

**Comment by detabc** [comment]
Just want to add my recent experience with the Pixel 5A and perhaps the latest GOS (not sure about the exact version, but I updated as soon as the update was available).
The phone is currently dead - black screen, doesn't react to power button / power button + volume down / charger plugged in.

The point I wanted to make is following - the phone (the screen) froze three times before it died. 
For a first time (yesterday) when watching video using Brave. Then after some attempts I managed to restart the phone.
This morning happened it for a second time. The phone (screen) froze, then I managed to restart it with a long press of the power button. And for a third time an hour later I wasn't that lucky anymore, the phone didn't (re)start.

Few days ago I got some error log after an update (sorry, not possible to get it anymore) and system pop-up messages saying that some app needs to restart.

Sorry for the vague fashion of the report, but the phone is dead and I am unable to grab any data from it.
I am pretty confident the issue is not hardware related, but has something to do with the previous freezes as they immediately led to the current condition. 
The battery should not be an issue, it was fully charged this morning. I am pretty sad.

**Comment by thestinger** [comment]
@detabc That's 100% guaranteed to be a hardware issue and not relevant to this.

**Comment by thestinger** [comment]
Is this still an issue with the November update?

**Comment by blizzardengle** [comment]
> Is this still an issue with the November update?

I returned my Pixel 8. @mattinahat13 ?

**Comment by mattinahat13** [comment]
Sorry, haven't been logged in, but yes, the issue is still persisting. Notably the phone is very stable on wi-fi. Tried removing the eSIM but still getting the issue. Been reflashing the OS/factory resets now and again, but can't pinpoint the exact cause for this issue.

**Comment by thestinger** [comment]
@mattinahat13 Please open a new issue since it's not necessarily the same problem.

**Comment by nobunagaPirate** [comment]
Im actually experiencing the same issue described. Even after yesterday's update. I posted on the forum, and it seemed to be an upstream bug

https://discuss.grapheneos.org/d/9030-apps-crash-on-pixel-8-pro

Is there anything i can do to help debug this? As opposed to OP, returning the phone is not really an option and id like to have a working phone.

**Comment by thestinger** [comment]
Please open a new issue.
